### ISSUE TITLE
Can't get PUT to work with --id

### ISSUE THREAD

**Issue by markandersonus** [typicode/json-server#684]
POST works OK and creates a new note with noteId = 6
when I try 
PUT /notes/6 and a JSON body, I get 404 Not found

my command line is

json-server --watch db.json --id noteId --routes routes.json  --port 10005

Routes how

{
"/api/*": "/$1",
"/notes": "/notes",
"/notes/:noteId": "/notes?noteId=:noteId",
"/notes/resource/:resourceId": "/notes?resourceId=:resourceId",
"/getShiftNotes": "/notes",
"/getJobNotes/:jobNo": "/notes?jobNo=:jobNo",
"/getNotesByTaskId/:taskId": "/notes?taskId=:taskId"
}

**Comment by Nilegfx** [comment]
Man, Your problem is not in the `noteId` with put (this works just fine), your problem is in the rewrites that you added, looks like you add a bunch of useless rewrites that changes the behaviour of calling resource. 

I also feel that you are confused between PUT, PATCH and POST methods, please read more about the reset APIs [here](http://www.restapitutorial.com/lessons/httpmethods.html)

a quick note, when you put, you should provide the resource id `/notes/1`

also have a look at my code comments in [this answer ](https://github.com/typicode/json-server/issues/690#issuecomment-348616467) before this block of code 

```js
SERVER.use(JSON_SERVER.rewriter({
  '/home/:id': '/home?id=:id'
}));
```

simply, clean the rewrites as the following: 
```js
{
"/getShiftNotes": "/notes",
"/getJobNotes/:jobNo": "/notes?jobNo=:jobNo",
"/getNotesByTaskId/:taskId": "/notes?taskId=:taskId"
}
```

**Comment by markandersonus** [comment]
thanks for the reply. I did resolve the problem. 

I don't think I am confusing this. I posed a note and got ID. Later I am replacing the entire resource

I'm also puzzle by you suggestion to "clean" my rewrite. (getShiftNotes IS wrong in my example) I need all these URI's except the getShiftNotes. Your "cleaning" has removed URI's that I need

I'm most puzzled by removal of this one. 

"/notes/:noteId": "/notes?noteId=:noteId",

without it, isn't it impossible to get a note by noteId? Maybe I'm missing a fundamental point

thanks for taking time to respond

Regards

mark
